qCC/ccGraphicalSegmentationTool.cpp

```diff
@@ -953,18 +987,36 @@ void ccGraphicalSegmentationTool::segment(bool keepPointsInside, ScalarType clas
 					pointInside = CCCoreLib::ManualSegmentationTools::isPointInsidePoly(P2D, m_segmentationPoly);
 				}
 
-				if (classifSF)
+				if (classifSF) // classification mode
 				{
-					// classification mode
 					if (pointInside)
 					{
 						classifSF->setValue(i, classificationValue);
 					}
 				}
-				else
+				else // standard segmentation mode
 				{
-					// standard segmentation mode
-					visibilityArray[i] = (keepPointsInside != pointInside ? CCCoreLib::POINT_HIDDEN : CCCoreLib::POINT_VISIBLE);
+					unsigned char visibility = (keepPointsInside != pointInside ? CCCoreLib::POINT_HIDDEN : CCCoreLib::POINT_VISIBLE);
+					visibilityArray[i] = visibility;
+
+					if (cloneSelection) // specify an output visibility array for the export mode (exported points will be hidden until the Segment tool is closed)
```

Comment:
I don't understand why you need to use the 'segment' method when exporting the current selection? (this the visibility array is already initialized on each entity that needs one, and you can query it with 'getVisibilityArray'.

Why don't you simply add a dedicated 'export' method that goes through the entities and calls createNewCloudFromVisibilitySelection or createNewMeshFromSelection?

Reply:
All the machinery with the omp stuff of the selection tool is in the segment method, I did not want to duplicate too much code. It is right that the code is a little bit complicated. What is your thoughts?

Reply:
So my understanding is that you use this button to export the current selection. So I was expecting the visibility array to be already up to date (no need to call the 'segment' method again. If the selection is not done, which means the user has not chosen whether the points should be inside or outside the polygon, I would find the workflow a bit clunky.

So for me, the visibility array should be already set properly, and you don't need to call 'segment' again. Am I missing something?

Reply:
In my use case, I want to export several small patches and I click on the export button each time I close a poly-line. So I activate the segment tool each time. What is practical is that I can see which patches have been exported as they are removed from the screen and I can also pause the segmentation to move my cloud. 
It is very practical when I want to select patches for manual classification. Once I have all my patches, I quit the segmentation tool and my patches are displayed (thanks to m_enableOnClose, m_disableOnClose).

Reply:
How do you choose whether to export inside or outside then?

My general comment is that the way you use this tool is not clearly expressed by a simple 'export/clone' button.

Reply:
I thought that the use of the dolly sheep was sufficiently meaningful to say that you will duplicate the selection (inside). The original segmentation tool allows you to split into several parts, to remove points (segment out), but not to copy the points you selected without modifying the initial cloud.
Maybe something like copy/paste could be more clear?

Reply:
So my issue is that it's really counter intuitive (for me) that this option applies to what's inside the polygon only but not yet segmented. It's almost a different tool in its spirit. The '#' icon (classification) was already a bit weird in this sense (even though the fact that the points are not disappearing makes things clearer).

I think we could compromise with an icon to the right of the '#' icon (instead of to the left), and that would look like this:
![export](https://github.com/CloudCompare/CloudCompare/assets/2735509/647aa2e2-c6d4-41db-b5d7-1308365eecf6)

And in the code, don't lump the standard segmentation mode and the export mode together, but make them 2 clearly separated 'else' cases. See my other comment.